packages/core/src/index.ts

```diff
@@ -1,89 +1,185 @@
+import { Zencode } from "@restroom-mw/zencode";
+import { BlockContext } from "./block-context";
 const functionHooks = initHooks;
 
 export default async (req: Request, res: Response, next: NextFunction) => {
+
   if (req.url === "/favicon.ico") {
     return;
   }
 
-  const getContractOrFail = async (name: string) => {
-    try {
-      return Zencode.byName(name, ZENCODE_DIR);
-    } catch (err) {
-      if (err.code === "ENOENT") {
-        const message = await getMessage(req);
-        res.status(404).send(message);
-      }
-    }
-  };
-
   const runHook = (hook: string, args: any) => {
     try {
       return getHooks(hook, res, args);
     } catch (e) {
       sendError(`[EXCEPTION IN REGISTERED HOOK ${hook}]`, e);
     }
   };
-
-  const sendError = (subject: string, e: Error = null) => {
+  /**
+   * Centralized api error handling
+   * @param {subject} string subject 
+   * @param {e} NodeJS.ErrnoException error
+   */
+  const sendError = (subject: string, e: NodeJS.ErrnoException = null) => {
     const exception = e ? e.stack || e.message : "";
     const message = subject + "\n\n\n" + exception;
-    if (!res.headersSent) {
-      res.status(500).json({
-        zenroom_errors: zenroom_errors,
-        result: zenroom_result,
-        exception: message,
+    if (e.code === "ENOENT") {
+      getMessage(req).then( (mes)=>{
+        res.status(404).send(mes);
       });
-      if (e) next(e);
+    } else{
+    if (!res.headersSent) {
+        res.status(500).json({
+          zenroom_errors: zenroom_errors,
+          result: zenroom_result,
+          exception: message,
+        });
+        if (e) next(e);
     }
+   }
   };
 
+  /**
+   * Centralized api response handling
+   * @param {restroomResult} RestroomResult containing restroom result 
+   * @param {res} Response endpoint response
+   */
+  const buildEndpointResponse = (restroomResult: RestroomResult, res: Response) => {
+    if (restroomResult?.error) {
+      sendError(restroomResult.errorMessage, restroomResult.error);
+    } else {
+      res.status(restroomResult.status).json(restroomResult?.result);
+    }
+  };
+
+  async function resolveRestroomResult(restroomResult: RestroomResult): Promise<RestroomResult> {
+    return new Promise((resolve) => {
+      resolve(restroomResult);
+    });
+  };
+
+  /**
+   * Function responsible to execute the chain
+   * @param {ymlFile} string containing restroom result 
+   * @param {data} object data object coming from endpoint 
+   * @param {keys} object keys object coming from file 
+   */
+  async function executeChain(fileContents: string, data: any): Promise<any> {
+    const ymlContent: any = yaml.load(fileContents);
+    const startBlock: string = ymlContent.start;
+  
+    return await evaluateBlock(startBlock, ymlContent, data);
+  }
+
+  const getRestroomResult = async (contractName:string, data:any) : Promise<RestroomResult> => {
+    const isChain = contractName.split(".")[1] === 'chain' || false;
```

Comment:
is possible to have the contract name desinence `chain` configurable via an env variable in the same flavor of `ZENCODE_DIR` and make the default to `.chain`?
90% of cases are covered since, probably chains will be created from the UI, but configurability also needeeeeed 😆